src/components/filter/Filter.tsx

```diff
@@ -1,153 +1,163 @@
-                  setFilters={setFilters}
-                  {...item}
-                  data={item.data as DropdownProps[]}
+      <Wrapper style={[withSearch && styles.noPadding]}>
+        <TouchableOpacity
+          onPress={() => setIsCollapsed(!isCollapsed)}
+          accessibilityLabel={`${isCollapsed ? texts.filter.showFilter : texts.filter.hideFilter} ${
+            a11yLabel.button
+          }`}
+          style={styles.button}
+        >
+          <BoldText small primary={!isCollapsed}>
+            {isCollapsed ? texts.filter.showFilter : texts.filter.hideFilter}
+          </BoldText>
+          <Icon.Filter
+            size={normalize(16)}
+            style={styles.icon}
+            color={!isCollapsed ? colors.primary : colors.darkText}
+          />
+        </TouchableOpacity>
+
+        {isOverlay ? (
+          <Modal
+            animationType="slide"
+            onRequestClose={() => setIsCollapsed(!isCollapsed)}
+            presentationStyle="pageSheet"
+            visible={!isCollapsed}
+          >
+            <Header
+              backgroundColor={colors.transparent}
+              centerComponent={{
+                text: texts.filter.header,
+                style: { color: colors.darkText, fontSize: normalize(18), fontWeight: '700' }
+              }}
+              rightComponent={{
+                icon: 'close',
+                color: colors.darkText,
+                onPress: () => setIsCollapsed(!isCollapsed)
+              }}
+            />
+            <Divider />
+            <Wrapper style={styles.noPaddingBottom}>
+              <FilterComponent
+                filters={filters}
+                filterTypes={filterTypes}
+                setFilters={setFilters}
+              />
+            </Wrapper>
+
+            <Wrapper style={[styles.noPaddingTop, styles.alignLeft]}>
+              <WrapperRow style={{ gap: normalize(16) }}>
+                <Button
+                  disabled={!!isNoFilterSet}
+                  invert
+                  notFullWidth
+                  onPress={() => {
+                    setIsCollapsed(!isCollapsed);
+                    setQueryVariables(initialFilters);
+                  }}
+                  title={texts.filter.resetFilter}
                 />
-              )}
-
-              {item.type === FILTER_TYPES.SUE.STATUS && (
-                <StatusFilter
-                  filters={filters}
-                  setFilters={setFilters}
-                  {...item}
-                  data={item.data as StatusProps[]}
+                <Button
+                  disabled={!!isNoFilterSet}
+                  notFullWidth
+                  onPress={() => {
+                    setIsCollapsed(!isCollapsed);
+                    setQueryVariables((prev) => ({ search: prev.search || '', ...filters }));
+                  }}
+                  title={texts.filter.filter}
                 />
-              )}
+              </WrapperRow>
```

Comment:
i have adjusted the layout a bit according to the mockups

<img src="https://github.com/user-attachments/assets/0d0f419e-b7f1-4c83-96db-6c7d7127d588" alt="Image Description" width="370">